### ISSUE TITLE
The multitouch extension ignore area has no effect

### ISSUE THREAD

**Issue by ryanoboril** [pqrs-org/Karabiner-Elements#2676]
I want to use this extension to make the stupidly huge touch pad area smaller.  I thought this extension would allow me to set an area that will ignore touches, but this seems to have no effect when I set it up.  There is another issue about this but it's closed and I don't believe it's getting attention, so I will create a new issue.  Here is the text of the original creator of the original issue that was ignored.  I'm using macOS Catalina 10.15.7

"Maybe I do not understand something properly. I expect areas on my touchpad to be completely ignored after having set ignored areas in Karabiner-Multitouch Extension. This is not the case here. The extension's GUI shows that I am moving the mouse cursor within the ignored area, but the cursor behaves totally normally, and I can also tap (which is what I really want to prevent because I often get cursor jumps when typing).

Any idea what the problem is? Or am I expecting something that the software is actually not expected to do?"

**Comment by tekezo** [comment]
The ignore area only takes effect on the first touch.
That is, if your finger touches the ignored area, it will be ignored, as intended.

Once a finger enters a valid area, it will remain valid even if it enters an ignored area.

I guess that this issue is opened due to the latter movement made you feel that the ignore area was not working correctly.

**Comment by dongheechoi** [comment]
I feel the same way as @ryanoboril does. I want this problem would be fixed. My mac is an M1 mac mini (2020), and I'm using Big Sur 11.5. And also I don't think this problem is because of what @tekezo said.

**Comment by Edgecrusherr160** [comment]
I'm having the same issue on my MacBook Pro (16-inch, 2019) running the latest 10.15.7. Going to have to upgrade it to macOS 12 soon, for work, but I doubt that will fix it.

I made sure to only click directly into the ignored area, but it has no affect. My computer even has SIP disabled, and set to allow all software. I'm trying to ignore the bottom portion. I even set it up to 20, just to test.

![Screen Shot 2021-11-19 at 10 03 50 a](https://user-images.githubusercontent.com/66455026/142644414-711201e9-925b-42d1-8f4f-0fbb4d629e6b.jpg)
.

**Comment by switz** [comment]
Same issue here -- this feature would be life changing. The trackpad is way too large.

**Comment by blubdog** [comment]
It's not working on my new 2021 Macbook Pro M1 Max (Monterey).  First-touch does not work either.

**Comment by mrfour0004** [comment]
Same issue on my 2019 Macbook Pro 16", I didn't see any effect after setting the ignored area🥲.

**Comment by prutschman** [comment]
I'm also seeing no effect from setting an exclusion region. Here's a screen recording showing exactly what actions I was performing:

https://youtu.be/pTnD9fV8ZJA

Karabiner MultiTouch v14.3.0, macOS 12.1 on 14" M1 Pro (MacBookPro18,4)

Here I perform a click-and-drag to move a window. I set a very large ignore area to make it easier to demonstrate, but the problem also occurs with a small area.

First, I use the allowed region of the trackpad to click and drag the window. This works as expected.

Second, I use the ignored area of the trackpad to click and drag the window. I expected this to be rejected, but it unexpectedly allowed me to move the window

**Comment by macieksmola** [comment]
Same here, MacBook Pro 2015 + Magic Trackpad 2.

**Comment by safris** [comment]
For me as well. Clicking directly on the Ignored Area, the click is still registered as normal.

Karabiner MultiTouch v14.10.0, macOS 12.6 on MacBook Pro M1 (16-inch, 2021).

**Comment by blubdog** [comment]
Still not working.  2021 Macbook Pro M1 Max running Ventura 13.1.  Just updated to Karabiner-Elements-14.11.0.  Uninstalled/reinstalled/rebooted...

**Comment by safris** [comment]
Still not working for me also: Karabiner-Elements-14.11.0 on 2021 MBP M1 Max.

**Comment by evgeniy-arapov** [comment]
14.12.0 Still not working for me on 2021 MBP M1 Max.

**Comment by olosegres** [comment]
All of us M1 16 Pro users are suffering from too big trackpad and this bug, please consider fixing it as soon as possible.

**Comment by tigakub** [comment]
I am having the same problem. It worked the first time I installed and activated it, but now it doesn't. MBA M2 14.3.1.

**Comment by skylatian** [comment]
can confirm on m2 air, ignore area has no effect

**Comment by teqtic** [comment]
I too am looking for a way to reduce the touch area on my 2021 16" MacBoko Pro's trackpad, but I think the original issue here may have missed the point. According to the documentation, "Karabiner-MultitouchExtension allows you to change keys only when your finger is on the trackpad." I'm not sure it's supposed to cause all input from the chosen areas of the trackpad to be ignored system-wide. It just causes the keys to not change when the ignored areas are touched for Karabiner purposes. Am I right?

**Comment by skylatian** [comment]
I ended up writing a VERY beta program that simulates a three-button touchpad on a mac trackpad. 

I could use the same logic to allow you to resize the touchpad area - would you be interested? My code is a bit buggy, but if you’re just looking to resize the touchpad area I could probably simplify it when I have time in the next few weeks. What kind of features are you looking for?

**Comment by teqtic** [comment]
@skylatian yes, I'm definitely very interested! I write Android apps so I can try to help develop it as well. I just had no idea where to start. How do you intercept the trackpad touch inputs at a system wide level? Please email me or open a project where we can discuss this. You can contact me at teqtic at gmail dot com. Thanks!!!

**Comment by skylatian** [comment]
@teqtic After exploring my old code, I don't believe I'll be able to block cursor _movement_, but I absolutely can ignore clicks, taps, etc. Does that satisfy what you're after? If so, I'll make a project on github (and link it here). I'm rebuilding my other project from the ground up for my own sanity, and this is an easy addition!

**Comment by teqtic** [comment]
@skylatian Yes, that should still help since at least unintended touches will be blocked. What prevents you from blocking cursor movement? Please have a look at the strategy used in [this](https://github.com/cocoahuke/shrink_trackpad) project as well. There is an [open issue](https://github.com/cocoahuke/shrink_trackpad/issues/10) that it does not work on Apple silicon, but I am about to try anyway using [this guide](https://www.sweetwater.com/sweetcare/articles/kernel-extensions-on-mac-with-apple-silicon/) for loading custom kexts on Apple silicon.

**Comment by skylatian** [comment]
Sounds good! I started a discussion here, let's move there to avoid cluttering up this issue further.

https://github.com/skylatian/tbtouchpad_shrink/discussions/2
